pkg/minikube/cruntime/docker.go

```diff
@@ -531,19 +537,56 @@ func (r *Docker) setCGroup(driver string) error {
 	if driver == constants.UnknownCgroupDriver {
 		return fmt.Errorf("unable to configure docker to use unknown cgroup driver")
 	}
+	var (
+		containsCGroupName bool
+		writeToDockerdJSON bool
+		daemonData         map[string]any
+	)
+	natCgroupDriver := "native.cgroupdriver"
+	cgroupKeyValue := natCgroupDriver + "=" + driver
 
-	klog.Infof("configuring docker to use %q as cgroup driver...", driver)
-	daemonConfig := fmt.Sprintf(`{
-"exec-opts": ["native.cgroupdriver=%s"],
-"log-driver": "json-file",
-"log-opts": {
-	"max-size": "100m"
-},
-"storage-driver": "overlay2"
-}
-`, driver)
-	ma := assets.NewMemoryAsset([]byte(daemonConfig), "/etc/docker", "daemon.json", "0644")
-	return r.Runner.Copy(ma)
+	file, err := os.ReadFile(DockerDaemonJSON)
+	if err != nil {
+		klog.Warningf("Failed to load %s. Creating new json file.", DockerDaemonJSON)
+		daemonStr := &DockerDaemon{
+			Exopt: []string{cgroupKeyValue},
+		}
+		newDaemonFile, _ := json.MarshalIndent(daemonStr, "", " ")
+		ma := assets.NewMemoryAsset(newDaemonFile, "/etc/docker", "daemon.json", "0644")
+		return r.Runner.Copy(ma)
+	}
+
+	// we don't know exact structure that user has already been using
+	err = json.Unmarshal(file, &daemonData)
+	if err != nil {
+		klog.Warningf("Failed to parse %s", file)
+	}
+	execOpts := daemonData["exec-opts"]
+	if execOpts == nil {
+		klog.Infof("%s does not contain any 'exec-opts' key. Configuring docker to use %s as cgroup driver...", DockerDaemonJSON, driver)
+		daemonData["exec-opts"] = []string{cgroupKeyValue}
+		writeToDockerdJSON = true
+	} else {
+		klog.Infof("%s contains exec-opts option. Checking for %s key presence.", DockerDaemonJSON, natCgroupDriver)
+		jsonExecOpts := execOpts.([]interface{})
+		for _, value := range jsonExecOpts {
+			dKey := strings.Split(value.(string), "=")[0]
+			if dKey == natCgroupDriver {
+				containsCGroupName = true
+			}
+		}
+		if !containsCGroupName {
+			klog.Infof("Insert %s driver into %s file.", natCgroupDriver, DockerDaemonJSON)
+			daemonData["exec-opts"] = append(jsonExecOpts, cgroupKeyValue)
+			writeToDockerdJSON = true
+		}
```

Comment:
```suggestion
				return nil
			}
		}
		klog.Infof("Insert %s driver into %s file.", natCgroupDriver, DockerDaemonJSON)
		daemonData["exec-opts"] = append(jsonExecOpts, cgroupKeyValue)
		writeToDockerdJSON = true
```
If Cgroup is already expect can just return and save some nesting.